packages/components/src/utils/font-values.js

```diff
@@ -1,14 +1,9 @@
-const DEFAULT = {
-	fontFamily:
+export default {
+	'default.fontFamily':
 		"-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
-	fontSize: '13px',
-};
+	'default.fontSize': '13px',
```

Comment:
👋 @saramarcondes  I'm curious about these changes and what motivated them…

Other "values" utilities use nested objects, e.g.

https://github.com/WordPress/gutenberg/blob/7778a3630356c3a26c8e91a1bba98f7bacbb3e80/packages/components/src/utils/colors-values.js#L118-L121
```js
color( 'blue.wordpress.700' );
```

These rely on lodash [`get`](https://lodash.com/docs/4.17.15#get)'s string object-path-like property access (`get( { foo: { bar: 'baz' } }, 'foo.bar' )`). In this case, it looks like the object-path-like strings are used directly as the keys. This was a bit surprising to me 🙂 

That's not necessarily a bad thing! it's just inconsistent and the keys here like `helpText.fontSize` were surprising to me.

We may be able to move away from `get` and rely more on TypeScript to ensure we're accessing values that exist.

Reply:
My intention was to keep things type-check-able! `get` obscures everything, the way I’ve set things up you get nice autocomplete. I chose to namespace values using the dot notation to match that nice feature from the colors utility while still getting the typecheck benefits.

Is it possible for us to maintain a “getter” function while having truly nested objects?

Reply:
Just to be clear, I think this direction is great but we can go farther 🙂 


> My intention was to keep things type-check-able! `get` obscures everything

`get` will actually do a pretty good job ([it's definition is interesting to look at](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/51b221165695d0d644867d37853168482e1c33ac/types/lodash/common/object.d.ts#L1023-L1101)), but JS/JSDoc is likely hindering it. Type arguments don't seem to work as well 😢 

> Is it possible for us to maintain a typechecked “getter” function while having truly nested objects?

I'm confident we could do a couple of levels of nesting without much trouble. Arbitrary depth is at least more complicated (we'd probably take [inspiration from `get`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/51b221165695d0d644867d37853168482e1c33ac/types/lodash/common/object.d.ts#L1023-L1101))

However, we know statically everything about these objects! What purpose does the getter serve? We swap parens for braces and [we have the same behavior](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAYgeQHIBUDKMC8MDeAoASAHIATAUwDMBDAVwBsoA6C8KOKgWwEs6BPIgFyECAIgC0VAA6S6ZMRF7QyHADQwAQnS5gA1gFkqwNIqjK4rNUTRkA5iDIwAqgEkiagEogARiCgg1CAAevDZkYGJoVJBqjl40YFA0agDCUVBUAE5kdHSWABLZAG5kUFzAVDBIZDRkbjAQURDyZBlcFCIqhKSUtAzMrGhcAF61AjBEAIwAzJKBbnhdABbZkihkgUwsCYMjguMTAEyz84Qc3jxkaxt62ubbw2RjkwBsx50AvjBUEDCgkFAAbgWeHWkhAGVgFHiwFK4BgWygAB4ANIwdamMAkH46Mi8EAUGBQXiSMj4+DIdAAPgAFDAcbwxqiAJRjIkksmIVBoADayIAurg8DAYFlEhkwOSudz6Xy8O8Fn9oPCsPDWNSiMs6Kt0f17rsmXhFZCAPoqznobkalZXTYDB5EPlAA).

```js
const fontSize = font( 'helpText.fontSize' );
const fontSize_ = FONTS[ 'helpText.fontSize' ];
```

If we want nested objects and we accept that a getter isn't necessary (it may be, please convince me) then we just access them the `.` notation and can get excellent type support:

![Screen Shot 2020-10-30 at 14 25 42](https://user-images.githubusercontent.com/841763/97710324-e9e15f80-1abb-11eb-96bb-3d52e31548c9.png)

[Here's a playground link with some of this set up already](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBDYA7AngYwgG2nAZlCEOAckVQ2ygCZiBuAKAyQGd4AxAeQDkAVAZTgBeOAG96cCSQAmAU1wBDAK6YYAOlwQkMNvJDBMKYgC5xkswCIAtPLBhMMy8xSsZIADRwAQpmQBrALLyaHzOMK5smjAexHwyAOYQMnAAqgCSxB4AShAARhAwEB4cAB4ocTJIlnzyLB7JOYpaih4AwjUw8lAymJjRABLdAG4yiGjycFwyijIZcMw1zI4yUMC45m6mEsSyCspqGlp8wABeM0YkAIwAzGDFGfSbJAAW3WA8MsX7kUenxpdUt3ujxAuX0Mnen38yAihxOMnOxAuADZARsAL5weTMOBMVgMB4fSCwPCNNCITR4SIAHgA0nAPmEkFJsb4ZCgILgECgwDIOXBOLw+AA+AAUrJQ5xpAEpzjBubzOQL+ABtGkAXVEjy6MEUUCQ-O4KvFGoA-Ca4OYAMQABlt5noaIeuPgBxgPySwldIuIL0wbwZ6m+cOIUsYmlYlNhpwA+kIDYLlT7XhCvlGZmqHvRCdAXaTyfqoHEcvIRS9igA1eSYaZx4izKtgJ7jYQXKWaszO0RwKAeOIeHJwDHCMjoLDQUsfSvVmRS1QFTJFkVShhmbW6-UAA0LxZFABIRFA0R593Ej3B9zkz-uG020VKNwxHQTikT4J3PABBPgAUTjYjMOSYEEvgIjatr3GYADuTzAGEoG4Ah9wYliOLhjA+IAPQAFRYeIWFwDwHAACIcOcLSRMgNYFHAihgFI8hhHAgxVtM2IoeYADiVDmHAsjMMAcT6jIgxYMMzCqHheEWlxPFdLgyzYtRMAvHAILDCAFTwNR8hSMxSBoEkylJMgYRQAoBlwHyRlwIBEBoL49JSLB0ASXA+FPDAMBgMwRgYRhcSwU8ig5KoGAgBhADq0BSAACl0zDMP5iiMjkyxxBhwAJaxGEXAAHEiSIAOx4RhWYvjmqEsPAXF-o8gHTOc-5mJIGlOYoICNY8zWSBoaCKD5JBgYVaDFhB3XNb1-VEZ0IGDQhuBjd1R6PMtZhxFA8gSu240AJy2qBFwyIdh2Lc1hX7YNhUAKzXddsx+XA-gyCM2IACyqEiRgXAgDKVTAG0RvIcTyMgEbQbBMiueNSIXcQlo7a9CMI-dGGPc9MDYlcX0pKkllQHAQGFoZv1MP9WLwEDINVXA4NhFD3WvbDlpoCzp1mFQTNSFzKMpMwMhSJS+MghGeRQLIUDiV1kgXEzuDWnLcunatkj0VAvgcRtW1Nd1YArCAnRbXDx1HTMGxmMrEitcA7Ua5tnXjT9nygbdN1XUrZuSD4cQebbWtSxIijAKBSOI69bOSHzTCq4bzOsx73WmYgBugTIhUyLlMg7e7DqYtizr4tmxKdkRH6ZDS0YcZkH4AJq1WYe3WgdO2HVQVynblTOt1QuVUFIp3nY3g1XK3hVXGgp0w4PcOM69SKvX38dwFdTNXavUgFTzyR8wgMHYsL8BWSpiAaYLvEzXAEMgJLZiM1Plow0i8ij6dVxM0iaCFYVuUXDzAAyAkeRkBGdam0d4MRxDUGySR+r81Ph+D8js3yRABmoR4HM75p1yjtHabdF4XGXqBXKUgdqP2tH-ABYRgGazAW+SBqUaJ8wFhofG8C4BIE0JYMInw-ooPpnAGWd9cq4B2rleQP9F52CCDIJ4WBxbnG3CWLiqgQEoGVA3NUHhrQfSoFKdEudKp4mfK+Ax8AS5l2jBwGKH4WipB4LXYQ2s4AN3kUWEscNbTWiujLWYWidq6MeB3QeCjvRgQ8fIHxqhcpXX8WYAeLidzuIftaHIETcoxMkJPeJbjQnN1wXALR110kSGXkE1xITrRiJ7lnTRqhCpFLgLfLJ5TKlUGqfkj60TF6v1KQkm0R00BUFNu0pEOjF7oKaUbEZr0rhuxqavepAiJmWioK3V60yImdLqsBdaEBGhSDYPoTATSzHl0rjXZUA8NHtLqXolC+cjEVWLqXculjrG2OrtGX+qQOJ9B4HXSQzjuxlLhq-NZq8ImvU2WYQJSzXpUCuvlPJWjXr1LiUC3pBUv75QiTM+pmT0XZMKjkfK8gaA1KuPUkpSzm47UKqSiJ8L6mNIJSEnaV02U7XHjU0ZjxulLKqYjJFqh8H1PGSyuGuViU5BIRE1seCLrBLhoVV6yrlURM8f45Cec0IF3KkXNCcAvk-J4BXKu9jtoSEBXDUl8h5A5DJQEpmOQro5BZg62JTMWZSGtFIMOi98VwykIVeishTpUsGjIQZr1U48w4hACATCcDmBUfzHil82JMhstFBSfDmVw3TjIeQMgUkeAenGhNp9zBdB0poAwaakBgGSngYA3RmSYkzcwHkaBgBVkQXMboMgyTAE0HwvlEadiKzGbLK4uBXq4FmY8RZc0hE7WEadSRBkZGYDkSyz8P5VC0xkDUpEmytUmN1cYzsRrfkWKsTYuxnzvm-P+ZahVZS93fgPTBMIcz6kwt3V+T9h6amQtRW+ncH6v0QxA3i8DJZIPAfabixe4bgkIe-UepDTK4MinQ9B9pjKuk4bwz+gjoriOAag6RrRIr5U9Pg5RxDNH6nFnsjsvZByehNOvSas51dlTdKuVotJtztVVUwqjD8XNYLDqQL2ig0BJYPT4IQJInQkj4EIHADyXkfJ+X1qyA90VdZAPEtAdKfEBJIAwk2JkeQIC+AwhoPZDFZOJQUxLDCEkyqXoNZ4X+yRfwOMeJBYz8UBqOI9XfW0BUdolpWovVWs1IsAqZraQqFwxHZzMFbdq9tdppetCerB4cJD-vcdaUenLStwDRRVwq5kasBrAkQ1KNXw0VfkD6914082WiuFcHIVwpDxYdmOuGBU0BIhkH6-2cAxVwxyPJBruC5tLvzVdedCgaswW9l7DyoEhup3kjVyaA16sjXCYvZaZ77k+ceQaj8v9vyZD+cFswKBujYEgvBZJuVEanS6FIUCxC51ztOutZ6SBQKvTtflJESF9F3cLkg6mH4iL+FSFwF9O9XDwjgBuZiUARRwEsJYSCYBrBSD0JUIyGlLAeePCIfzgWjNixMwlC5to1R3g3IvOnMhppqxlucQnnQSdk4p1TmnnCXj0485YJLyxLAyyZyzz9uWQCBj6swXn6IJMTF0EAsAUiBbMWnNiZh3Z5CQVQpQbEZZUJ8Xu-q6maQccC-OOjzHXA5xy6PVstjBA9nnBIwHgC2zg9MiIplO19hgeGqfbxs1yp0EaLqtmqA5wlEqK59adPAFM8RB117jHWO-d46F74VXGexbLBj-MQC-Ns9UGUZrZUt8C+SFFuLf+3sYAt7b5tVP3PF5AVSkcuAJzTXnJKV3iQXCYAN7j83qfzyZ-8fwfn-nDIq+h8Yxhnfnw+8Ha8JRwCwFRPnoea7iMLQOC-w4JkAQ73JCqHfx+xLM1fbnA0kTEUIgZ40+fGHgSiSWvs9SSWHAJu3aco5w0+ry961cX+as0BQQsEKAJ+A+a+5iiB7yj6xqi8muP+cASixBmsi8Kig+KiY+FCJBf+5QABZ4PGG+oBre+2MAEBtB-evsWB5wLBeBD6PGi89U+ODBMgTBHg6ubBqgoh9SVYyw2BT2L2PAi8OkNOpePui8gc5waQV+d29A9+j+z+qggcshkeuyTI9AQAA).

Reply:
Certainly, we could just access the object directly and that would work fine. However @ItsJonQ has some good rationale for a "getter" function laid out here: https://g2components.wordpress.com/2020/10/23/creating-a-variable-system/

Because of that we have some limitations as to what information we can give to TypeScript, at least that's my understanding. The definition for `get` you linked is indeed interesting, however it only works if you split the path into segments rather than using a the dotted notation, which is already prevelant in the codebase for the existing getter functions.

Given those limitations and given that we probably want to be able to namespace things, I did the best with what I had... perhaps there's still something I'm missing that would allow us to avoid this style of namespacing while keeping the getter functions?